tpm2/structures.go

```diff
@@ -90,18 +90,21 @@ func (p Public) Encode() ([]byte, error) {
 	return concat(head, params)
 }
 
-func decodePublic(in *bytes.Buffer) (Public, error) {
+// DecodePublic decode a TPMT_PUBLIC message. No error is returned if
+// the input has extra trailing data.
+func DecodePublic(in []byte) (Public, error) {
+	buf := bytes.NewBuffer(in)
```

Comment:
Name bytes.Buffer `in` to minimize the diff

Reply:
Will do!